platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java

```diff
@@ -24,13 +24,13 @@
  *
  * @param softwareVersion
  * 		The integer value of the software version of the AddressBookTestingToolMain SwirldMain application.
- * @param stakingBehavior
- *      The integer value of the staking behavior of the AddressBookTestingToolState SwirldState.
+ * @param weightingBehavior
+ *      The integer value of the weighting behavior of the AddressBookTestingToolState SwirldState.
  * @param testScenario
- *     The integer value of the test scenario being run for validation.
+ *     The string value of the test scenario being run for validation. This must match an enumerated value in {@link AddressBookTestScenario}.
  */
 @ConfigData("addressBookTestingTool")
 public record AddressBookTestingToolConfig(
         @ConfigProperty(defaultValue = "1") int softwareVersion,
-        @ConfigProperty(defaultValue = "0") int stakingBehavior,
-        @ConfigProperty(defaultValue = "0") int testScenario) {}
+        @ConfigProperty(defaultValue = "0") int weightingBehavior,
+        @ConfigProperty(defaultValue = "SKIP_VALIDATION") String testScenario) {}
```

Comment:
Once https://github.com/hashgraph/hedera-services/issues/6106 is fixed we can easily use the enum type directly. Maybe you can add that as a comment for future.

Reply:
added comment.